### PR TITLE
[P4-1739] Create interruption card support for Person Escort Record framework

### PR DESCRIPTION
**Pull request by teneightfive**
### What changed

Adds support to use custom step types from the frameworks. This initial piece of work supports
using the [Interruption card](https://home-office-digital-patterns.herokuapp.com/patterns/interruption-card) pattern to direct users to things they must carry out offline.

To support this, it also involved adding markdown support to this PR.

### Why did it change

One of the journeys within the Person Escort Record requires the user to take action outside
of the digital service. The interruption card pattern allows us to describe those instructions in
a clear way for now. 

### Issue tracking

- [P4-1739](https://dsdmoj.atlassian.net/browse/P4-1739)

## Screenshots

<kbd></kbd>

#### Automated testing

- [x] Added unit tests to cover changes
- [ ] Added end-to-end tests to cover any journey changes

#### Manual testing

Has been tested in the following browsers:

- [x] Chrome
- [x] Firefox
- [ ] Edge
- [ ] Internet Explorer

### Environment variables

- [x] No environment variables were added or changed

### Other considerations

Commit messages with a `fix` or `feat` type are automatically used to generate the [changelog](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/CHANGELOG.md) and
[GitHub release notes](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/releases) during the release task. Please make sure they will read well on their own in a
summary of changes and that the commit body gives a more detailed description of those changes if necessary.

- [x] No new [Personally Identifiable Information (PII)](https://support.google.com/analytics/answer/6366371?hl=en) is being sent via analytics
- [x] Update [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/README.md) with any new instructions or tasks

### CODE REVIEW

File: nodemon.json [hunk 1]
```diff
@@ -6,11 +6,10 @@
     "config/",
     "locales/",
     "server.js",
-    "start.js"
-  ],
-  "ignore": [
-    ".build/"
+    "start.js",
+    ".build/frameworks/**/*.json"
```
Comment (solidgoldpig) [inline nodemon.json:10]: Is it worth adding `.env` too?

File: app/person-escort-record/views/form-step.njk [hunk 2]
```diff
@@ -1,15 +1,19 @@
 {% extends "form-wizard.njk" %}
 
+{% set mainClasses = "interruption-card" if options.stepType == 'interruption-card' %}
```
Comment (solidgoldpig) [inline app/person-escort-record/views/form-step.njk:3]: Just a thought - is it worth having a `form-step` and `form-interruption-card` template?

File: nodemon.json (hunk 1 above)
Comment (teneightfive) [reply nodemon.json:10]: Not sure just restarting the server would be enough would it? Wouldn't you need to reload you shell session too?

File: app/person-escort-record/views/form-step.njk (hunk 2 above)
Comment (teneightfive) [reply app/person-escort-record/views/form-step.njk:3]: Not a bad thought. Although would mean updating the render method to handle changing of the template and probably now having 3 templates (one base to inherit from, plus one for the two _types_).

Maybe one to consider if the type becomes more drastically different in what it renders but in this case it's only applying a class and hiding the save action.

What do you think?

**Pull request merged**